wrappers/python/nntile/layer/llama_attention.py

```diff
@@ -893,6 +905,443 @@ def forward_async(self):
             self.out_proj_bias.value.wont_use()
         self.y.value.wont_use()
 
+    def _forward_mlp_q_dynamic(self, x):
+        q_partial_tr_bt_shape = tuple(
+            self.q_transposed.value.basetile_shape[:-2]
+        ) + tuple(x.shape[-2:])
+        q_partial_tr_shape = tuple(self.q_transposed.value.shape[:-2]) + tuple(
+            x.shape[-2:]
+        )
+        q_partial_tr = nntc.empty(
+            q_partial_tr_shape,
+            basetile_shape=q_partial_tr_bt_shape,
+            dtype=type(x),
+        )  # (kv_group_size, n_head_kv, head_size, n_seq_dyn, n_batch_dyn)
+
+        q_partial_bt_shape = (
+            (self.q.value.basetile_shape[0],)
+            + tuple(x.shape[-2:])
+            + tuple(self.q.value.basetile_shape[-2:])
+        )
+        q_partial_shape = (
+            (self.q.value.shape[0],)
+            + tuple(x.shape[-2:])
+            + tuple(self.q.value.shape[-2:])
+        )
+        q_partial = nntc.empty(
+            q_partial_shape, basetile_shape=q_partial_bt_shape, dtype=type(x)
+        )  # (head_size, n_seq_dyn, n_batch_dyn, kv_group_size, n_head_kv)
+
+        # Q_transposed = einsum('ijkl,lmn->ijkmn', W_Q, X_Q_dyn)
+        # gemm (kv_group_size, n_head_kv, head_size, n_emb)
+        # by (n_emb, n_seq_dyn, n_batch_dyn)
+        # into (kv_group_size, n_head_kv, head_size, n_seq_dyn, n_batch_dyn)
+        gemm_async(
+            1.0,
+            notrans,
+            self.w_q.value,
+            notrans,
+            x,
+            0.0,
+            q_partial_tr,
+            1,
+            0,
+            redux=self.redux,
+        )
+
+        # Rotate axes into
+        # (head_size, n_seq_dyn, n_batch_dyn, kv_group_size, n_head_kv)
+        transpose_async(1.0, q_partial_tr, q_partial, 2)
+        q_partial_tr.invalidate_submit()
+
+        if self.in_proj_bias_q is not None:
+            # batched add_fiber (head_size, batch=(kv_group_size, n_head_kv))
+            # into
+            # (head_size, n_seq_dyn, n_batch_dyn, batch=(kv_group_size, n_head_kv)) # noqa: E501
+            add_fiber_async(1, self.in_proj_bias_q.value, 1, q_partial, 0, 2)
```

Comment:
make float parameters appear as 1.0 and 0.0

Reply:
done